include/aikido/constraint/outcome/CollisionFreeOutcome.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef AIKIDO_CONSTRAINT_COLLISIONFREEOUTCOME_HPP_
+#define AIKIDO_CONSTRAINT_COLLISIONFREEOUTCOME_HPP_
+
+#include <vector>
+#include "TestableOutcome.hpp"
+#include "dart/collision/CollisionObject.hpp"
+#include "dart/collision/Contact.hpp"
+#include "dart/dynamics/BodyNode.hpp"
+#include "dart/dynamics/ShapeFrame.hpp"
+#include "dart/dynamics/ShapeNode.hpp"
+
+namespace aikido {
+namespace constraint {
+
+class CollisionFreeOutcome : public TestableOutcome
+{
+public:
+  /// Documentation inherited.
+  bool isSatisfied() const override;
+
+  /// Returns a string with each pair of CollisionObject names on a separate
+  /// line. Each pair is also marked as being a normal collision or self
+  /// collision.
+  std::string toString() const override;
+
+  /// Store a Contact object from a pairwise collision. Used by CollisionFree
+  // to modify this object.
+  /// \param pairwiseContact Contact object to store.
+  void markPairwiseContact(const dart::collision::Contact& pairwiseContact);
+
+  /// Store a Contact object from a self collision. Used by CollisionFree to
+  /// modify this object.
+  /// \param selfContact Contact object to store.
+  void markSelfContact(const dart::collision::Contact& selfContact);
+
+  /// Return a copy of the vector storing the Contact objects from pairwise
+  /// collisions.
+  std::vector<dart::collision::Contact> getPairwiseContacts() const;
+
+  /// Return a copy of the vector storing the Contact objects from self
+  /// collisions.
+  std::vector<dart::collision::Contact> getSelfContacts() const;
+
+  /// Gets the name of a CollisionObject. The name returned is that of the
+  /// corresponding BodyNode (if possible). If not, the name of the ShapeFrame
+  /// is returned instead. This is a helper for toString().
+  /// \param object object pointer to CollisionObject we want the name of.
+  std::string getCollisionObjectName(
+      dart::collision::CollisionObject* object) const;
```

Comment:
Nit: It seems `object` is not changed by this function. It would be nice to change the type to `const` so that we could pass const collision object as well.